HashTagTokenizer.java

```diff
@@ -7,34 +7,52 @@ public static void main(String[] args) {
 		String hashTag = args[0];
 		String []dictionary = readDictionary("dictionary.txt");
 		breakHashTag(hashTag, dictionary);
+		
 	}
 
 	public static String[] readDictionary(String fileName) {
 		String[] dictionary = new String[3000];
 
 		In in = new In(fileName);
 
-		// Your code here
+		for(int i = 0 ; i < dictionary.length ; i++){
+			dictionary[i]= in.readLine();
+		}
 
 		return dictionary;
 	}
 
 	public static boolean existInDictionary(String word, String []dictionary) {
-		// Your code here
+		boolean exist = false;
+		for(int i = 0 ; i < dictionary.length ; i++){
+			if(word.equals(	dictionary[i])){
+				exist = true; 
+				break;
+			}
+		}
+		return exist;
 	}
 
 	public static void breakHashTag(String hashtag, String[] dictionary) {
-
 		// Base case: do nothing (return) if hashtag is an empty string.
         if (hashtag.isEmpty()) {
             return;
         }
+		else{
+			hashtag = hashtag.toLowerCase();
+		}
```

Comment:
When there is `return` within if condition, there's no need to mention `else` since when the condition fulfilled it anyway won't reach the rest code.
```suggestion
			hashtag = hashtag.toLowerCase();
```